### PR TITLE
[MM-55455] Add "no-constant-binary-expression" eslint rule

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Adds 'no-constant-binary-expression' rule in mattermost/eslint. I feel it is an important rule, and that's why it included eslint package instead of webapp eslint config.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55455

#### Screenshots
n/a

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by hmhealey — APPROVED** [review]
Looks good assuming the isHidden change doesn't break anything. It seems like that setting is supposed to be visible normally

**Review by yasserfaraazkhan — APPROVED** [review]

File: webapp/channels/src/components/admin_console/admin_definition.tsx [hunk 1]
```diff
@@ -6934,7 +6934,7 @@ const AdminDefinition: AdminDefinitionType = {
                                     </ExternalLink>
                                 ),
                             },
-                            isHidden: true || it.not(it.licensedForFeature('SAML')),
+                            isHidden: it.not(it.licensedForFeature('SAML')),
```
Comment (jespino) [inline webapp/channels/src/components/admin_console/admin_definition.tsx:6937]: This is changing the previous existing behavior, that was intentional to disable this section from the admin console. Unless we have a good reason to renable it I wouldn't do this change.

**Review by jespino — CHANGES_REQUESTED** [review]
I added a comment about something that is changing the current behavior, otherwise looks good to me.

File: webapp/channels/src/components/admin_console/admin_definition.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/admin_console/admin_definition.tsx:6937]: I see if it was intentionally hidden then i will remove the complete field it if required in future we can always get it back from the commit history

**Review by jespino — APPROVED** [review]

**Pull request merged**